src/T2DMap.cpp

```diff
@@ -723,8 +723,14 @@ inline void T2DMap::drawRoom(QPainter& painter, QFont& roomVNumFont, QFont& mapN
     roomNameRectangle = roomRectangle.adjusted(-2000, realHeight, 2000, realHeight);
 
     painter.save();
-    painter.setFont(mapNameFont);
-    roomNameRectangle = painter.boundingRect(roomNameRectangle, Qt::TextSingleLine|Qt::AlignTop|Qt::AlignCenter, pRoom->name);
+    if (showRoomName) {
```

Comment:
It might also help to determine if there is a room name to be displayed - not all MUDs use names IIRC....

Reply:
Already implemented; I found another two optimizations.

Empty room names however are a non-problem in general, as rooms without names are unlikely to have their display-the-name flag set in the first place :-P